src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -213,7 +217,7 @@ internal class InProcOrRemoteHostAnalyzerRunner
             // handling of cancellation and exception
             var version = await DiagnosticIncrementalAnalyzer.GetDiagnosticVersionAsync(project, cancellationToken).ConfigureAwait(false);
 
-            using var reader = ObjectReader.TryGetReader(stream, leaveOpen: true, cancellationToken);
+            using var reader = ObjectReader.TryGetReader(stream, leaveOpen: false, cancellationToken);
```

Comment:
why did this change?  was it a bug before? consider doc.

Reply:
it wasn't a bug before - the method was called in a very different stream pattern.

Reply:
I'm gonna add an abstraction for the streaming API pattern.